packages/core/src/PartialExecutionUtils/cleanRunData.ts

```diff
@@ -0,0 +1,26 @@
+import type { INode, IRunData } from 'n8n-workflow';
+import type { DirectedGraph } from './DirectedGraph';
+
+/**
+ * Returns new run data that does not contain data for any node that is a child
+ * of any start node.
+ * This does not mutate the `runData` being passed in.
+ */
+export function cleanRunData(
+	runData: IRunData,
+	graph: DirectedGraph,
+	startNodes: INode[],
+): IRunData {
+	const newRunData: IRunData = { ...runData };
```

Comment:
Is a shallow clone safe here? I.e. we don't need deep clone?

Reply:
I kind wish we'd use something like https://immutable-js.com/ or similar 😄

Reply:
Yeah I was asking because I'm not at all familiar how the execution data works. Your answer makes sense since we are only removing certain properties here, so all good 👍